tests/tensor/signal/test_conv.py

```diff
@@ -0,0 +1,59 @@
+from functools import partial
+
+import numpy as np
+import pytest
+from scipy.signal import convolve as scipy_convolve
+
+from pytensor import function
+from pytensor.graph.basic import equal_computations
+from pytensor.tensor import matrix, vector
+from pytensor.tensor.signal.conv import convolve, convolve1d
+from tests import unittest_tools as utt
+
+
+@pytest.mark.parametrize("kernel_shape", [3, 5, 8], ids=lambda x: f"kernel_shape={x}")
+@pytest.mark.parametrize("data_shape", [3, 5, 8], ids=lambda x: f"data_shape={x}")
+@pytest.mark.parametrize("mode", ["full", "valid", "same"])
+def test_convolve1d(mode, data_shape, kernel_shape):
+    data = vector("data")
+    kernel = vector("kernel")
+    op = partial(convolve1d, mode=mode)
+
+    rng = np.random.default_rng()
+    data_val = rng.normal(size=data_shape).astype(data.dtype)
+    kernel_val = rng.normal(size=kernel_shape).astype(kernel.dtype)
+
+    fn = function([data, kernel], op(data, kernel))
+    np.testing.assert_allclose(
+        fn(data_val, kernel_val),
+        scipy_convolve(data_val, kernel_val, mode=mode),
+    )
+    utt.verify_grad(op=lambda x: op(x, kernel_val), pt=[data_val])
+
+
+def test_convolve1d_batch():
+    x = matrix("data")
+    y = matrix("kernel")
+    out = convolve1d(x, y)
+
+    # Convolution is unchanged by order
```

Comment:
I don't understand the comment

Reply:
Doesn't matter if x, y, or y,x. I'm using that property to confirm batching is working as expected

Reply:
Clarified comment